x-pack/plugins/ml/public/application/jobs/jobs_list/components/ml_job_editor/ml_job_editor.tsx

```diff
@@ -7,10 +7,9 @@
 import React, { FC } from 'react';
 
 import { EuiCodeEditor, EuiCodeEditorProps } from '@elastic/eui';
-import { expandLiteralStrings } from '../../../../../../shared_imports';
-import { xJsonMode } from '../../../../components/custom_hooks';
+import { expandLiteralStrings, XJsonMode } from '../../../../../../shared_imports';
 
-export const ML_EDITOR_MODE = { TEXT: 'text', JSON: 'json', XJSON: xJsonMode };
```

Comment:
I presume it's expensive to initialize a new instance of XJsonMode, isn't it? that's why I imported a reference

Reply:
I don't think it is that expensive, perhaps just not necessary.

The init behaviour should be the same as your code, I just removed the `x-pack/plugins/ml/public/application/components/custom_hooks/use_x_json_mode.ts` file where it was being instantiated and moved `new XJsonMode()` here. It doesn't look like the hook was being used.